pkg/cluster/cluster.go

```diff
@@ -877,9 +877,16 @@ func (c *Cluster) initSystemUsers() {
 			c.Spec.ConnectionPooler = &acidv1.ConnectionPooler{}
 		}
 
-		username := util.Coalesce(
-			c.Spec.ConnectionPooler.User,
-			c.OpConfig.ConnectionPooler.User)
+		// Using superuser as pooler user is not a good idea. First of all it's
+		// not going to be synced correctly with the current implementation,
+		// and second it's a bad practice.
+		username := c.OpConfig.ConnectionPooler.User
+
+		if c.Spec.ConnectionPooler.User != c.OpConfig.SuperUsername {
```

Comment:
Does it make sense to avoid names from `protected_role_names` and `replication_username` as well ?

Reply:
Good point. Yes, I believe it makes sense.